pyanaconda/modules/boss/install_manager/install_manager.py

```diff
@@ -0,0 +1,264 @@
+        self._subscriptions.append(s)
+
+        s = self._actual_task.ErrorRaised.connect(self._task_error_raised)
+        self._subscriptions.append(s)
+
+    def _disconnect_task(self):
+        for subscription in self._subscriptions:
+            subscription.disconnect()
+
+    def _test_if_running(self, log_msg=None):
+        if self._actual_task is not None:
+            return True
+        else:
+            log.warn(log_msg)
+            return False
+
+    def _task_running_changed(self, is_running):
+        if not is_running:
+            self._tasks_done_step += self._actual_task.ProgressStepsCount
+            if self._tasks:
+                self._actual_task = self._tasks.pop()
+                self._run_task()
+            else:
+                log.info("Installation finished.")
+                self._actual_task = None
+                self._install_running_signal.emit(False)
+
+    def _task_error_raised(self, error_description):
+        self._error_raised_signal.emit(error_description)
+
+    @property
+    def installation_running(self):
+        """Installation is running right now.
+
+        :returns: True if installation is running. False otherwise.
+        """
+        return self._actual_task is not None
+
+    @property
+    def task_name(self):
+        """Get name of the running task."""
+        if self._test_if_running("Can't get task name when installation is not running."):
+            return self._actual_task.Name
+        else:
+            return ""
+
+    @property
+    def task_description(self):
+        """Get description of the running task."""
+        if self._test_if_running("Can't get task description when installation is not running."):
+            return self._actual_task.Description
+        else:
+            return ""
+
+    @property
+    def progress_steps_count(self):
+        """Sum of steps in all tasks used for installation."""
+        if self._test_if_running("Can't get sum of all tasks when installation is not running."):
+            return self._step_sum
+        else:
+            return 0
+
+    def _progress_changed(self, step, msg):
+        actual_progress = step + self._tasks_done_step
+        self._progress_changed_signal.emit(actual_progress, msg)
+        self._progress_changed_float_signal.emit(actual_progress / self._step_sum, msg)
+
+    @property
+    def progress(self):
+        """Get progress of the installation.
+
+        :returns: (step: int, msg: str) tuple.
+                  step - step in the installation process.
+                  msg - short description of the step
+        """
+        if self._test_if_running("Can't get task progress when installation is not running."):
+            (step, msg) = self._actual_task.Progress
+            actual_progress = step + self._tasks_done_step
+
+            return actual_progress, msg
+        else:
+            return 0, ""
+
+    @property
+    def progress_float(self):
```

Comment:
Couldn't be the floating progress part of the regular progress?

Reply:
I don't think there is a reason to use both in one place ever. So you will use only the one you want to use.

Reply:
Ok.